### PR TITLE
add command whitelist for non-interactive shells

### PR DESCRIPTION
**Pull request by Huuancao**

### CODE REVIEW

**Comment by nimdanitro** [comment]
thanks, can you also update the doc/log-user-session.pod?

File: src/log-user-session.c [hunk 1]
```diff
@@ -495,8 +505,24 @@ void write_log(int fd_input, int fd_log, char *buffer, size_t size) {
...
 }
 
+int is_command_whitelisted(const char *original_command) {
+    // retrieve first token which is the command
+    char *command = strtok(strdup(original_command), " \n\t");
+
+    int i;
+    for (i = 0; i < *whitelist_size ; i++) {
+        if ( !strcmp(command, opt_command_whitelist[i]) ) {
+            free(command);
```
Comment (nimdanitro) [inline src/log-user-session.c:515]: I don't understand why you have to free command here

**Review by nimdanitro — APPROVED** [review]

File: src/log-user-session.c (hunk 1 above)
Comment (nimdanitro) [reply src/log-user-session.c:515]: alright.. strdup allocates mem.. and should be freed.

**Comment by Huuancao** [comment]
> thanks, can you also update the doc/log-user-session.pod?

Sure, do you have any default whitelisted commands?

**Review by nvinzens — APPROVED** [review]

File: src/log-user-session.c (hunk 1 above)
Comment (kbucheli) [reply src/log-user-session.c:515]: There is a memory leak if the command is not whitelisted

**Review by nimdanitro — APPROVED** [review]

**Review by kbucheli — APPROVED** [review]

File: doc/log-user-session.pod [hunk 2]
```diff
@@ -112,6 +112,13 @@ C<LogNonInteractiveData = 1>
 
 Log also data of non-interactive sessions (by default on).
 
+
+=head2 NonInteractiveCommandWhitelist
+
+C<NonInteractiveCommandWhitelist = rsync,ps,scp>
+
+Log only the command of the aforementioned whitelisted commands and not the output (by default empty).
```
Comment (kbucheli) [inline doc/log-user-session.pod:120]: Maybe also state that this is only relevant if LogNonInteractiveData is on.

**Pull request merged**